inspirehep/records/serializers/bibtex.py

```diff
@@ -0,0 +1,168 @@
+            "editor",
+            "address",
+            "organization",
+            "pages",
+        ],
+        "misc": ["howpublished", "reportNumber", "author"],
+        "mastersthesis": ["reportNumber", "school", "address", "type", "author"],
+        "proceedings": [
+            "publisher",
+            "series",
+            "number",
+            "volume",
+            "reportNumber",
+            "editor",
+            "address",
+            "organization",
+            "pages",
+        ],
+        "book": [
+            "publisher",
+            "isbn",
+            "author",
+            "series",
+            "number",
+            "volume",
+            "edition",
+            "editor",
+            "reportNumber",
+            "address",
+        ],
+        "inbook": [
+            "chapter",
+            "publisher",
+            "author",
+            "series",
+            "number",
+            "volume",
+            "edition",
+            "editor",
+            "reportNumber",
+            "address",
+            "type",
+            "pages",
+        ],
+        "article": ["author", "journal", "number", "volume", "reportNumber", "pages"],
+    }
+
+    def __init__(self, schema_class=BibTexCommonSchema):
+        self.schema_class = schema_class()
+
+    def create_bibliography_entry(self, record):
+        bibtex_document_type = self.schema_class.get_bibtex_document_type(record)
+
+        data = self.schema_class.dump(record).data
+        doc_type = data.pop("doc_type", None)
+        texkey = data.pop("texkey", None)
+        fields = (
+            self.COMMON_FIELDS_FOR_ENTRIES
+            + self.FIELDS_FOR_ENTRY_TYPE[bibtex_document_type]
+        )
+        template_data = [
+            (key, str(value)) for key, value in data.items() if value and key in fields
+        ]
+        data_bibtex = [
+            texkey,
+            Entry(
+                doc_type,
+                template_data,
+                persons={
+                    "author": [
+                        Person(x)
+                        for x in BibTexCommonSchema.get_authors_with_role(
+                            data.get("authors", []), "author"
+                        )
+                    ],
+                    "editor": [
+                        Person(x)
+                        for x in BibTexCommonSchema.get_authors_with_role(
+                            data.get("authors", []), "editor"
+                        )
+                    ],
+                },
+            ),
+        ]
+        return data_bibtex
+
+    def create_bibliography(self, record_list):
+        bib_dict = {}
+        for record in record_list:
+            texkey, entries = self.create_bibliography_entry(record)
+            bib_dict[texkey] = entries
+
+        bib_data = BibliographyData(bib_dict)
+        writer = BibtexWriter()
+        return writer.to_string(bib_data)
+
+    def serialize(self, pid, record, links_factory=None):
```

Comment:
`pid` and `links_factory` are not used here

Reply:
is the invenio signature for serializers https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/serializers/json.py#L36-L47